src/runtime.js

```diff
@@ -151,6 +151,7 @@ let tornado = {
    */
   exists(val, placeholderNode, bodies, context) {
     if (this.util.isPromise(val)) {
+      placeholderNode = this.insertPendingBody(placeholderNode, bodies.pending, context) || placeholderNode;
```

Comment:
```
{@existsWithPromise key=data}
   show when done
{:pending}
  <div class="pending"></div>
{/existsWithPromise}
```

It'll be the same logic. What if any runtime level feature be a helper and every compile level feature is in the grammar/compiler.